pkg/compiler/types.go

```diff
@@ -81,15 +84,37 @@ var typeInt = &typeDesc{
 	Check: func(comp *compiler, t *ast.Type, args []*ast.Type, base prog.IntTypeCommon) {
 		typeArgBase.Type.Check(comp, t)
 	},
+	CheckConsts: func(comp *compiler, t *ast.Type, args []*ast.Type, base prog.IntTypeCommon) {
+		if len(args) > 0 && len(args[0].Colon) != 0 {
```

Comment:
What if we have int[100, 200], it won't pass any verification?

Reply:
That's allowed? Won't it be a `const`? I couldn't find any constant expressed as an `int`, so I assumed it wasn't allowed by the syntax :/

Reply:
It should not be allowed. But as far as I see the new code won't reject it b/c len(args[0].Colon) == 0 in this case, so we will not do any checks whatsoever. We should.

Reply:
I added a check that the first int argument always is a range, i.e., contains a colon.